src/NzbDrone.Core/MediaFiles/DiskScanService.cs

```diff
@@ -131,6 +144,25 @@ public void Scan(List<string> folders = null, FilterFilesType filter = FilterFil
                 {
                     _logger.Debug("Specified scan folder ({0}) doesn't exist.", folder);
 
+                    if (_configService.CreateEmptyAuthorFolders)
+                    {
+                        if (_configService.DeleteEmptyFolders)
+                        {
+                            _logger.Debug("Not creating missing author folder: {0} because delete empty author folders is enabled", folder);
+                        }
+                        else
+                        {
+                            _logger.Debug("Creating missing author folder: {0}", folder);
+
+                            _diskProvider.CreateFolder(folder);
```

Comment:
This is the other part I have an issue with.  You're just creating whatever was passed in as `folders` if it doesn't exist I think.